### PR TITLE
[filebeat][netflow]: fix template sharing

### LINKED ISSUE #42080: Filebeat Netflow Template Sharing not working
During my tests with the Filebeat Netflow Input, I found that sharing templates by using the `share_templates` parameter did not work. The template was received multiple times, but was never applied to subsequent Netflow events.

```
"[netflow-v9] Packet from:127.0.0.1:59773 src:256 seq:499564737"
"[netflow-v9] FlowSet ID 0 length 80"
"[netflow-v9] state - addTemplate 256 -"
"[netflow-v9] Packet from:127.0.0.1:56714 src:256 seq:499564738"
"[netflow-v9] Packet from:127.0.0.1:54741 src:256 seq:499564739"
"[netflow-v9] FlowSet ID 256 length 100"
"[netflow-v9] No template for ID 256"
```

I then looked at the code and found out that the shareTemplates parameter is not set in the decoder. I created a Merge Request that may fixes this issue: https://github.com/elastic/beats/pull/42079

### PR DESCRIPTION
**Pull request by simioa**
## Proposed commit message

```
Pass the share_templates configuration option into the NetflowV9Protocol
struct. The parameter was not being set, and therefore was always false so
it was not possible to use this option.

Added a test case to prevent future regressions.

Closes #42080
```

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues

- Closes #42080

### CODE REVIEW

**Comment by simioa** [comment]
@andrewkroh I added Tests and the Changelog entry

File: x-pack/filebeat/input/netflow/decoder/v9/v9_test.go [hunk 1]
```diff
@@ -249,3 +249,59 @@ func TestCustomFields(t *testing.T) {
...
+		3, 4,
+	}
+
+	t.Run("Template sharing enabled", func(t *testing.T) {
+		sharedTemplates := true
+		key := MakeSessionKey(addr, sourceID, sharedTemplates)
+		cfg := config.Defaults()
+		cfg.WithSharedTemplates(sharedTemplates)
+		proto := New(cfg)
+		flows, err := proto.OnPacket(test.MakePacket(templatePacket), addr)
```
Comment (andrewkroh) [inline x-pack/filebeat/input/netflow/decoder/v9/v9_test.go:274]: I think we would want some kind of scenario that adds template data from a one address and then data from different address, and demonstrates that it produces a flow. This verifies that the template is shared.

Comment (simioa) [reply x-pack/filebeat/input/netflow/decoder/v9/v9_test.go]: That makes sense, I have updated the tests accordingly.

**Review by andrewkroh — APPROVED** [review]

**Comment by andrewkroh** [comment]
run docs-build

**Comment by andrewkroh** [comment]
Thanks @simioa 👍

**Comment by simioa** [comment]
Hi @andrewkroh 
is it possible to make a backport into 8.16 and 8.17 as well?

**Pull request merged**